sys/arm64/arm64/locore.S

```diff
@@ -109,7 +193,48 @@ _start:
 	mrs x2, cctlr_el0
 	orr x2, x2, #(CCTLR_PERMVCT_MASK | CCTLR_SBL_MASK)
 	msr cctlr_el0, x2
+
+#ifdef __CHERI_PURE_CAPABILITY__
+	/*
+	 * Enable capablity sealing for branch and link at EL1
+	 * Use PCC/DDC address interpretation.
+	 * Use DDC as base for adrdp.
+	 */
+	mrs	x2, cctlr_el1
+	bic	x2, x2, #(CCTLR_PCCBO_MASK | CCTLR_DDCBO_MASK | CCTLR_ADRDPB_MASK)
+	orr	x2, x2, #(CCTLR_SBL_MASK)
+	msr	cctlr_el1, x2
+
+	/* We assume that we enter here in a64 mode. */
+	bx	#4
+	.arch_extension c64
 #endif
+#endif
+.endmacro
+
+	.text
+	.globl _start
+_start:
```

Comment:
... but really we should be setting no64+a64c here

Reply:
Yeah it's probably best. Moving it there.